public/app/core/components/SharedPreferences/SharedPreferences.tsx

```diff
@@ -94,45 +110,44 @@ export class SharedPreferences extends PureComponent<Props, State> {
 
   async componentDidMount() {
     const prefs = await this.service.load();
-    const dashboards = await backendSrv.search({ starred: true });
-    const defaultDashboardHit: DashboardSearchHit = {
-      id: 0,
-      title: 'Default',
-      tags: [],
-      type: '' as DashboardSearchItemType,
-      uid: '',
-      uri: '',
-      url: '',
-      folderId: 0,
-      folderTitle: '',
-      folderUid: '',
-      folderUrl: '',
-      isStarred: false,
-      slug: '',
-      items: [],
-    };
+    const dashboards = (await backendSrv.search({ starred: true })) as DashboardSearchItem[];
```

Comment:
I don't see why this is necessary. 'Just' deprecate `id` on `DashboardSearchItem` and don't use it. I don't see the need to create a new type without it just for the purpose of a migration.

```
export interface DashboardSearchHit extends DashboardSectionItem, DashboardSection, WithAccessControlMetadata {
  /** @deprecated */
  id: number;
}
```

will do the trick, or deprecate it on `DashboardSectionItem` and/or `DashboardSection`.

I think this is even better because it gives us the visible signal in the rest of the code that the field shouldn't be used.

Reply:
@joshhunt if I do that, I will need to add `id` when creating an object with the type`DashboardSearchHit`. Like [here](https://github.com/grafana/grafana/pull/51919/files#diff-008573da83c75271f854ceb9dd97c97c231e3ace2c93bc9a6b364ccfb14e73ceR119-R132). The other option is to define id as optional, but then I need to manage all logic around `id` anywhere is used. It is going to be a long migration, so we cannot modify them at once

Reply:
@ivanortegaalba  I think it's okay to leave it in there, if it's still on the interface. Once we've migrated all our usage away from it and we're no longer using it, then it's a lot simpler to then just remove the few remaining instances like this one in one PR.

I think it's important to not add in more instances of these patterns we don't want to continue as a social signal that we don't like writing code like this. If it feel it's an absolute _must_ (which I don't think it is 😅 ) then it should be ignorred using a `// eslint-ignore-next-line` comment so we accept the "smell" of this not being our preferred pattern.

Reply:
Agreed! I'll keep this in mind while completing the migration. Thanks a lot! 🙏